firmware/Core/Inc/adcs_drivers/adcs_types.h

```diff
@@ -0,0 +1,521 @@
+	uint8_t last_id;
+	bool processed:1; // 1-bit bool
+	enum ADCS_Error_Flag error_flag;
+	uint8_t error_index;
+} ADCS_CMD_Ack_Struct;
+
+typedef struct ADCS_ID_Struct {
+	uint8_t node_type;
+	uint8_t interface_version;
+	uint8_t major_firmware_version;
+	uint8_t minor_firmware_version;
+	uint16_t seconds_since_startup;
+	uint16_t ms_past_second;
+} ADCS_ID_Struct;
+
+typedef struct ADCS_Boot_Running_Status_Struct {
+    ADCS_Reset_Cause reset_cause;
+    ADCS_Boot_Cause boot_cause;
+    uint16_t boot_counter;
+    ADCS_Running_Program boot_program_index;
+    uint8_t major_firmware_version;
+    uint8_t minor_firmware_version;
+} ADCS_Boot_Running_Status_Struct;
+
+typedef struct ADCS_Comms_Status_Struct {
+    uint16_t cmd_counter;
+    uint16_t tlm_counter;
+    bool cmd_buffer_overrun:1; // 1-bit bool
+    bool i2c_tlm_error:1; // 1-bit bool
+    bool i2c_cmd_error:1; // 1-bit bool
+} ADCS_Comms_Status_Struct;
+
+typedef struct ADCS_Angular_Rates_Struct {
+    int32_t x_rate_milli_deg_per_sec;
+    int32_t y_rate_milli_deg_per_sec;
+    int32_t z_rate_milli_deg_per_sec;
+} ADCS_Angular_Rates_Struct;
+
+typedef struct ADCS_LLH_Position_Struct {
+    int32_t latitude_milli_deg;
+    int32_t longitude_milli_deg;
+    int32_t altitude_meters;
+} ADCS_LLH_Position_Struct;
+
+typedef struct ADCS_Power_Control_Struct{
+	ADCS_Power_Select cube_control_signal;
+	ADCS_Power_Select cube_control_motor;
+	ADCS_Power_Select cube_sense1;
+	ADCS_Power_Select cube_sense2;
+	ADCS_Power_Select cube_star_power;
+	ADCS_Power_Select cube_wheel1_power;
+	ADCS_Power_Select cube_wheel2_power;
+	ADCS_Power_Select cube_wheel3_power;
+	ADCS_Power_Select motor_power;
+	ADCS_Power_Select gps_power;
+} ADCS_Power_Control_Struct;
+
+typedef struct ADCS_Set_Unix_Time_Save_Mode_Struct{
+	bool save_now:1; // 1-bit bool
+	bool save_on_update:1; // 1-bit bool
+	bool save_periodic:1; // 1-bit bool
+	uint8_t period;
+} ADCS_Set_Unix_Time_Save_Mode_Struct;
+
+typedef struct ADCS_Orbit_Params_Struct {
+    double inclination;
+    double eccentricity;
+    double ascending_node_right_ascension;
+    double perigee_argument;
+    double b_star_drag_term;
+    double mean_motion;
+    double mean_anomaly;
+    double epoch;
+} ADCS_Orbit_Params_Struct;
+
+typedef struct ADCS_Rated_Sensor_Rates_Struct {
+	int32_t x_milli_deg_per_sec; 
+	int32_t y_milli_deg_per_sec;
+	int32_t z_milli_deg_per_sec; 
+} ADCS_Rated_Sensor_Rates_Struct;
+
+typedef struct ADCS_Wheel_Speed_Struct {
+    // TODO: Add bool for whether it's commanded or actual wheel speed
+	int16_t x; 
+	int16_t y;
+	int16_t z; 
+} ADCS_Wheel_Speed_Struct;
+
+typedef struct ADCS_Magnetorquer_Command_Struct {
+	int32_t x_ms; 
+	int32_t y_ms;
+	int32_t z_ms; 
+} ADCS_Magnetorquer_Command_Struct;
+
+typedef struct ADCS_Raw_Magnetometer_Values_Struct {
+	int16_t x; 
+	int16_t y;
+	int16_t z; 
+} ADCS_Raw_Magnetometer_Values_Struct;
+
+typedef struct ADCS_Fine_Angular_Rates_Struct {
+	int16_t x_milli_deg_per_sec; 
```

Comment:
Please indent all files using spaces and not tabs.

Reply:
This is done.